checker/src/org/checkerframework/checker/lock/LockAnnotatedTypeFactory.java

```diff
@@ -96,28 +106,374 @@ public LockTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, LockS
+                }
+            }
+
+            if (count == 0) {
+                return defaults.applyUnannotatedDefaults(element) ?
+                    SideEffectAnnotation.MAYRELEASELOCKS :
+                    SideEffectAnnotation.RELEASESNOLOCKS;
+            }
+
+            if (count > 1 && issueErrorIfMoreThanOnePresent) {
+                // TODO: Turn on after figuring out how this interacts with inherited annotations.
+                // checker.report(Result.failure("multiple.sideeffect.annotations"), element);
+            }
+
+            // If at least one side effect annotation was found, return the weakest.
+            for(int i = 0; i < countSideEffectAnnotations; i++) {
+                if (sideEffectAnnotationPresent[i]) {
+                    return SideEffectAnnotation.values()[i];
+                }
+            }
+        }
+
+        // When there is not enough information to determine the correct side effect annotation,
+        // return the weakest one.
+        return SideEffectAnnotation.MAYRELEASELOCKS;
+    }
+
+    private static class AnnotationPair { // Same contents as in AnnotatedTypeFactory.java
+        public final AnnotationMirror a1;
+        public final AnnotationMirror a2;
+        private int hashCode = -1;
+
+        public AnnotationPair(AnnotationMirror a1, AnnotationMirror a2) {
+            this.a1 = a1;
+            this.a2 = a2;
+        }
+
+        @Pure
+        @Override
+        public int hashCode() {
+            if (hashCode == -1) {
+                hashCode = 31;
+                if (a1 != null)
+                    hashCode += 17 * AnnotationUtils.annotationName(a1).toString().hashCode();
+                if (a2 != null)
+                    hashCode += 17 * AnnotationUtils.annotationName(a2).toString().hashCode();
+            }
+            return hashCode;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (!(o instanceof AnnotationPair))
+                return false;
+            AnnotationPair other = (AnnotationPair)o;
+            if (AnnotationUtils.areSameIgnoringValues(a1, other.a1)
+                    && AnnotationUtils.areSameIgnoringValues(a2, other.a2))
+                return true;
+            if (AnnotationUtils.areSameIgnoringValues(a2, other.a1)
+                    && AnnotationUtils.areSameIgnoringValues(a1, other.a2))
+                return true;
+            return false;
         }
 
+        @SideEffectFree
+        @Override
+        public String toString() {
+            return "AnnotationPair(" + a1 + ", " + a2 + ")";
+        }
+    }
+
+    @Override
+    protected void annotateImplicit(Tree tree, AnnotatedTypeMirror type,
+            boolean iUseFlow) {
+        super.annotateImplicit(tree, type, iUseFlow);
+        
+        if (tree.getKind() == Kind.METHOD_INVOCATION) {
+            // Check that matching @GuardSatisfied(index) on a method's formal return type/parameters matches
```

Comment:
This comment doesn't match what is happening.  I think this could is replacing the type's of arguments with @GBInaccessable if they will be assigned to a formal parameter with @GBSat.  The code looks good -- the comments just need to be update.